ksql-engine/src/main/java/io/confluent/ksql/util/PlanSummary.java

```diff
@@ -0,0 +1,148 @@
+import io.confluent.ksql.execution.plan.StreamGroupBy;
+import io.confluent.ksql.execution.plan.StreamMapValues;
+import io.confluent.ksql.execution.plan.StreamSelectKey;
+import io.confluent.ksql.execution.plan.StreamSink;
+import io.confluent.ksql.execution.plan.StreamSource;
+import io.confluent.ksql.execution.plan.StreamStreamJoin;
+import io.confluent.ksql.execution.plan.StreamTableJoin;
+import io.confluent.ksql.execution.plan.StreamWindowedAggregate;
+import io.confluent.ksql.execution.plan.TableAggregate;
+import io.confluent.ksql.execution.plan.TableFilter;
+import io.confluent.ksql.execution.plan.TableGroupBy;
+import io.confluent.ksql.execution.plan.TableMapValues;
+import io.confluent.ksql.execution.plan.TableSink;
+import io.confluent.ksql.execution.plan.TableSource;
+import io.confluent.ksql.execution.plan.TableTableJoin;
+import io.confluent.ksql.execution.plan.WindowedStreamSource;
+import io.confluent.ksql.execution.plan.WindowedTableSource;
+import io.confluent.ksql.execution.streams.StepSchemaResolver;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.schema.ksql.FormatOptions;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class PlanSummary {
+  private static final FormatOptions FORMAT_OPTIONS = FormatOptions.of(
+      IdentifierUtil::needsQuotes
+  );
+  private static final Map<Class<? extends ExecutionStep>, String> OP_NAME =
+      new ImmutableMap.Builder<Class<? extends ExecutionStep>, String>()
+          .put(StreamAggregate.class, "AGGREGATE")
+          .put(StreamWindowedAggregate.class, "AGGREGATE")
+          .put(StreamFilter.class, "FILTER")
+          .put(StreamFlatMap.class, "FLAT_MAP")
+          .put(StreamGroupBy.class, "GROUP_BY")
+          .put(StreamMapValues.class, "PROJECT")
+          .put(StreamSelectKey.class, "REKEY")
+          .put(StreamSink.class, "SINK")
+          .put(StreamSource.class, "SOURCE")
+          .put(StreamStreamJoin.class, "JOIN")
+          .put(StreamTableJoin.class, "JOIN")
+          .put(WindowedStreamSource.class, "SOURCE")
+          .put(TableAggregate.class, "AGGREGATE")
+          .put(TableFilter.class, "FILTER")
+          .put(TableGroupBy.class, "GROUP_BY")
+          .put(TableMapValues.class, "PROJECT")
+          .put(TableSink.class, "SINK")
+          .put(TableTableJoin.class, "JOIN")
+          .put(TableSource.class, "SOURCE")
+          .put(WindowedTableSource.class, "SOURCE")
+          .build();
+
+  private final QueryId queryId;
+  private final StepSchemaResolver schemaResolver;
+
+  public PlanSummary(final QueryId queryId, final KsqlConfig config, final MetaStore metaStore) {
+    this(
+        queryId,
+        new StepSchemaResolver(
+            Objects.requireNonNull(config, "config"),
+            Objects.requireNonNull(metaStore, "metaStore")
```

Comment:
nit: unnecessary null checks (the `StepSchemaResolver` does this in the constructor)